shared/util/pictures.js

```diff
@@ -9,128 +9,143 @@ type Info = {
   callbacks: Array<(username: string, urlMap: ?URLMap) => void>,
   done: boolean,
   error: boolean,
+  requesting: boolean,
 }
 
-// Done
-const _nameToURL: {[key: string]: ?Info} = {}
-// Not done
-const _pendingNameToURL: {[key: string]: ?Info} = {}
+const _nameToInfo: {[key: string]: ?Info} = {}
 
-const _getUserImages = throttle(() => {
-  const usersToResolve = Object.keys(_pendingNameToURL)
-  if (!usersToResolve.length) {
+const _getTeamPictures = (teamnames: Array<string>) => {
+  if (!teamnames.length) {
     return
   }
 
-  // Move pending to non-pending state
-  usersToResolve.forEach(username => {
-    const info: ?Info = _pendingNameToURL[username]
-    _nameToURL[username] = info
-    delete _pendingNameToURL[username]
+  teamnames.forEach(n => {
+    const info = _nameToInfo[n]
+    if (info) {
+      info.requesting = true
+    }
+  })
+
+  apiserverGetRpcPromise({
+    param: {
+      args: [
+        {key: 'team_names', value: teamnames.join(',')},
+        {key: 'formats', value: 'square_360,square_200,square_40'},
+      ],
+      endpoint: 'image/team_avatar_lookups',
+    },
   })
+    .then(response => {
+      JSON.parse(response.body).pictures.forEach((picMap, idx) => {
+        const teamname = teamnames[idx]
+        let urlMap = {
+          ...(picMap['square_200'] ? {'200': picMap['square_200']} : null),
+          ...(picMap['square_360'] ? {'360': picMap['square_360']} : null),
+          ...(picMap['square_40'] ? {'40': picMap['square_40']} : null),
+        }
+
+        const info = _nameToInfo[teamname]
+        if (info) {
+          info.done = true
+          info.requesting = false
+          info.urlMap = urlMap
+          info.callbacks.forEach(cb => cb(teamname, urlMap))
```

Comment:
below we do `const callbacks = info.callbacks; info.callbacks = []; callbacks.forEach(...)`. we should probably do that here, too. (Maybe make this a helper function?)